src/ncp/ncp_base_radio.cpp

```diff
@@ -403,6 +404,7 @@ otError NcpBase::DecodeStreamRawTxRequest(otRadioFrame &aFrame)
     aFrame.mInfo.mTxInfo.mMaxCsmaBackoffs     = OPENTHREAD_CONFIG_MAC_MAX_CSMA_BACKOFFS_DIRECT;
     aFrame.mInfo.mTxInfo.mMaxFrameRetries     = OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT;
     aFrame.mInfo.mTxInfo.mCsmaCaEnabled       = true;
+    aFrame.mInfo.mTxInfo.mIsHeaderUpdated     = false;
```

Comment:
Since we use the new `mIsHeaderUpdated` flag in `TxDone()` callbacks, we need to make sure that RCP report it back to host. I think this is missing in the current implementation. 

We have the change in `RadioSpinel` where we pass the flag down to the RCP for tx (from `SPINEL_CMD_PROP_VALUE_SET(SPINEL_PROP_STREAM_RAW)`) but we also need it to be passed as part of TX DONE event from RCP (in `NcpBase::LinkRawTransmitDone()` used on RCP) and then add code to parse it on host side in `RadioSpinel<InterfaceType, ProcessContextType>::HandleTransmitDone()`.

----

The more I read and think about it, the more I am convinced that adding `mIsHeaderUpdated` flag adds a lot of avoidable complexity with little benefit in return.

Reply:
Implemented.